src/SignalR/server/StackExchangeRedis/src/RedisHubLifetimeManager.cs

```diff
@@ -540,6 +655,31 @@ private async Task SubscribeToGroupAsync(string groupChannel, HubConnectionStore
         });
     }
 
+    private async Task SubscribeToReturnResultsAsync()
+    {
+        var channel = await _bus!.SubscribeAsync(_channels.ReturnResults(_serverName));
+        channel.OnMessage((channelMessage) =>
+        {
+            var completion = RedisProtocol.ReadCompletion(channelMessage.Message);
+            IHubProtocol? protocol = null;
+            foreach (var hubProtocol in _hubProtocolResolver.AllProtocols)
+            {
+                if (hubProtocol.Name.Equals(completion.ProtocolName))
+                {
+                    protocol = hubProtocol;
+                    break;
+                }
+            }
+            Debug.Assert(protocol is not null);
```

Comment:
Given this could in theory happen if you have servers running different versions of the app with different protocols, lets log.